kv/kvclient/kvcoord: TestMuxRangeFeedConnectsToNodeOnce failed

kv/kvclient/kvcoord.TestMuxRangeFeedConnectsToNodeOnce [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8582461?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8582461?buildTab=artifacts#/) on release-22.2 @ [9efe3c5c73d3a5eb54b3c4bfe2e3a83504702c39](https://github.com/cockroachdb/cockroach/commits/9efe3c5c73d3a5eb54b3c4bfe2e3a83504702c39):


```
=== RUN   TestMuxRangeFeedConnectsToNodeOnce
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/40ec041d13327ceafaefe721fe51f834/logTestMuxRangeFeedConnectsToNodeOnce2528852068
    test_log_scope.go:79: use -show-logs to present logs inline
*
* INFO: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
*
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1fc4783]

goroutine 7801 [running]:
github.com/cockroachdb/cockroach/pkg/roachpb.(*RangeFeedEvent).GetValue(...)
	github.com/cockroachdb/cockroach/pkg/roachpb/bazel-out/k8-fastbuild/bin/pkg/roachpb/roachpb_go_proto_/github.com/cockroachdb/cockroach/pkg/roachpb/api.pb.go:23214
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).singleRangeFeed(0xc001d20a00, {0x5215918?, 0xc002e5d380?}, {{0xc002e855d0, 0x5, 0x8}, {0xc0024f25e0, 0x5, 0x8}}, {0x174094c92a0d37d3, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:664 +0x1023
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).partialRangeFeed(0xc001d20a00, {0x5215918, 0xc002e5d380}, 0xc001528690, 0xc001c25310?, {{0xc002e855d0, 0x5, 0x8}, {0xc0024f25e0, 0x5, ...}}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:423 +0x8fb
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).RangeFeedSpans.func1.1({0x5215918?, 0xc002e5d380?})
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:207 +0x117
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x25
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:74 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:71 +0xa5
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMuxRangeFeedConnectsToNodeOnce.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-24179

kv/kvclient/kvcoord.TestMuxRangeFeedConnectsToNodeOnce [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8582462?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8582462?buildTab=artifacts#/) on release-22.2 @ [9efe3c5c73d3a5eb54b3c4bfe2e3a83504702c39](https://github.com/cockroachdb/cockroach/commits/9efe3c5c73d3a5eb54b3c4bfe2e3a83504702c39):


```
=== RUN   TestMuxRangeFeedConnectsToNodeOnce
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/40ec041d13327ceafaefe721fe51f834/logTestMuxRangeFeedConnectsToNodeOnce2965204857
    test_log_scope.go:79: use -show-logs to present logs inline
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1fca703]

goroutine 588173 [running]:
github.com/cockroachdb/cockroach/pkg/roachpb.(*RangeFeedEvent).GetValue(...)
	github.com/cockroachdb/cockroach/pkg/roachpb/bazel-out/k8-fastbuild/bin/pkg/roachpb/roachpb_go_proto_/github.com/cockroachdb/cockroach/pkg/roachpb/api.pb.go:23214
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).singleRangeFeed(0xc002e3a500, {0x521cb58?, 0xc000c6fc00?}, {{0xc00e49b228, 0x5, 0x8}, {0xc00d3dc370, 0x5, 0x8}}, {0x174095181df39505, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:664 +0x1023
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).partialRangeFeed(0xc002e3a500, {0x521cb58, 0xc000c6fc00}, 0xc0035cb720, 0xc0091c7f50?, {{0xc00e49b228, 0x5, 0x8}, {0xc00d3dc370, 0x5, ...}}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:423 +0x8fb
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).RangeFeedSpans.func1.1({0x521cb58?, 0xc000c6fc00?})
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:207 +0x117
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x25
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:74 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:71 +0xa5
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMuxRangeFeedConnectsToNodeOnce.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvclient/kvcoord.TestMuxRangeFeedConnectsToNodeOnce [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8582467?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8582467?buildTab=artifacts#/) on release-22.2 @ [9efe3c5c73d3a5eb54b3c4bfe2e3a83504702c39](https://github.com/cockroachdb/cockroach/commits/9efe3c5c73d3a5eb54b3c4bfe2e3a83504702c39):


```
=== RUN   TestMuxRangeFeedConnectsToNodeOnce
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/40ec041d13327ceafaefe721fe51f834/logTestMuxRangeFeedConnectsToNodeOnce540418459
    test_log_scope.go:79: use -show-logs to present logs inline
*
* INFO: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
*
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3222c5d]

goroutine 7991 [running]:
github.com/cockroachdb/cockroach/pkg/roachpb.(*RangeFeedEvent).GetValue(...)
	github.com/cockroachdb/cockroach/pkg/roachpb/bazel-out/k8-fastbuild/bin/pkg/roachpb/roachpb_go_proto_/github.com/cockroachdb/cockroach/pkg/roachpb/api.pb.go:23214
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).singleRangeFeed(0xc0029def00, {0x8c07a50, 0xc003927500}, {{0xc002f77358, 0x4, 0x8}, {0xc0033e1878, 0x5, 0x8}}, {0x17409747670c23d4, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:664 +0x153d
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).partialRangeFeed(0xc0029def00, {0x8c07a50, 0xc003927500}, 0xc0037190e0, 0xc003a23e20?, {{0xc002f77358, 0x4, 0x8}, {0xc0033e1878, 0x5, ...}}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:423 +0xb3d
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).RangeFeedSpans.func1.1({0x8c07a50, 0xc003927500})
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:207 +0x21f
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x4d
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:74 +0x87
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:71 +0x12f
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMuxRangeFeedConnectsToNodeOnce.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvclient/kvcoord.TestMuxRangeFeedConnectsToNodeOnce [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8589389?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8589389?buildTab=artifacts#/) on release-22.2 @ [9efe3c5c73d3a5eb54b3c4bfe2e3a83504702c39](https://github.com/cockroachdb/cockroach/commits/9efe3c5c73d3a5eb54b3c4bfe2e3a83504702c39):


```
=== RUN   TestMuxRangeFeedConnectsToNodeOnce
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/40ec041d13327ceafaefe721fe51f834/logTestMuxRangeFeedConnectsToNodeOnce596601831
    test_log_scope.go:79: use -show-logs to present logs inline
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1fc4783]

goroutine 7671 [running]:
github.com/cockroachdb/cockroach/pkg/roachpb.(*RangeFeedEvent).GetValue(...)
	github.com/cockroachdb/cockroach/pkg/roachpb/bazel-out/k8-fastbuild/bin/pkg/roachpb/roachpb_go_proto_/github.com/cockroachdb/cockroach/pkg/roachpb/api.pb.go:23214
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).singleRangeFeed(0xc000c5ef00, {0x5215918?, 0xc00233b500?}, {{0xc003113350, 0x5, 0x8}, {0xc0010bb1d8, 0x5, 0x8}}, {0x1740e36dce9d9822, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:664 +0x1023
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).partialRangeFeed(0xc000c5ef00, {0x5215918, 0xc00233b500}, 0xc00168ec30, 0xc005f8bfe0?, {{0xc003113350, 0x5, 0x8}, {0xc0010bb1d8, 0x5, ...}}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:423 +0x8fb
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).RangeFeedSpans.func1.1({0x5215918?, 0xc00233b500?})
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:207 +0x117
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x25
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:74 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:71 +0xa5
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #96558 kv/kvclient/kvcoord: TestMuxRangeFeedConnectsToNodeOnce failed [C-test-failure O-robot branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMuxRangeFeedConnectsToNodeOnce.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvclient/kvcoord.TestMuxRangeFeedConnectsToNodeOnce [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8589394?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8589394?buildTab=artifacts#/) on release-22.2 @ [9efe3c5c73d3a5eb54b3c4bfe2e3a83504702c39](https://github.com/cockroachdb/cockroach/commits/9efe3c5c73d3a5eb54b3c4bfe2e3a83504702c39):


```
=== RUN   TestMuxRangeFeedConnectsToNodeOnce
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/40ec041d13327ceafaefe721fe51f834/logTestMuxRangeFeedConnectsToNodeOnce538249959
    test_log_scope.go:79: use -show-logs to present logs inline
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3222c5d]

goroutine 7837 [running]:
github.com/cockroachdb/cockroach/pkg/roachpb.(*RangeFeedEvent).GetValue(...)
	github.com/cockroachdb/cockroach/pkg/roachpb/bazel-out/k8-fastbuild/bin/pkg/roachpb/roachpb_go_proto_/github.com/cockroachdb/cockroach/pkg/roachpb/api.pb.go:23214
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).singleRangeFeed(0xc002ce0a00, {0x8c07a50, 0xc0031480c0}, {{0xc0003d4678, 0x4, 0x8}, {0xc002b6aa48, 0x5, 0x8}}, {0x1740e44980fdec3e, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:664 +0x153d
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).partialRangeFeed(0xc002ce0a00, {0x8c07a50, 0xc0031480c0}, 0xc0044b0690, 0xc002b2f9b0?, {{0xc0003d4678, 0x4, 0x8}, {0xc002b6aa48, 0x5, ...}}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:423 +0xb3d
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).RangeFeedSpans.func1.1({0x8c07a50, 0xc0031480c0})
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:207 +0x21f
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x4d
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:74 +0x87
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:71 +0x12f
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #96558 kv/kvclient/kvcoord: TestMuxRangeFeedConnectsToNodeOnce failed [C-test-failure O-robot branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMuxRangeFeedConnectsToNodeOnce.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvclient/kvcoord.TestMuxRangeFeedConnectsToNodeOnce [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8596535?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8596535?buildTab=artifacts#/) on release-22.2 @ [94796359a4a73690b8e39484d8ecc515b64db77b](https://github.com/cockroachdb/cockroach/commits/94796359a4a73690b8e39484d8ecc515b64db77b):


```
=== RUN   TestMuxRangeFeedConnectsToNodeOnce
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/40ec041d13327ceafaefe721fe51f834/logTestMuxRangeFeedConnectsToNodeOnce116929578
    test_log_scope.go:79: use -show-logs to present logs inline
*
* INFO: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
*
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1fc4363]

goroutine 7687 [running]:
github.com/cockroachdb/cockroach/pkg/roachpb.(*RangeFeedEvent).GetValue(...)
	github.com/cockroachdb/cockroach/pkg/roachpb/bazel-out/k8-fastbuild/bin/pkg/roachpb/roachpb_go_proto_/github.com/cockroachdb/cockroach/pkg/roachpb/api.pb.go:23214
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).singleRangeFeed(0xc001817400, {0x5215318?, 0xc003d19b80?}, {{0xc005dddd98, 0x2, 0x8}, {0xc004311ec8, 0x3, 0x8}}, {0x1741325edf02fd8a, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:664 +0x1023
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).partialRangeFeed(0xc001817400, {0x5215318, 0xc003d19b80}, 0xc001dda050, 0xc0063d9a10?, {{0xc005dddd98, 0x2, 0x8}, {0xc004311ec8, 0x3, ...}}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:423 +0x8fb
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).RangeFeedSpans.func1.1({0x5215318?, 0xc003d19b80?})
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:207 +0x117
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x25
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:74 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:71 +0xa5
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #96558 kv/kvclient/kvcoord: TestMuxRangeFeedConnectsToNodeOnce failed [C-test-failure O-robot branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMuxRangeFeedConnectsToNodeOnce.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvclient/kvcoord.TestMuxRangeFeedConnectsToNodeOnce [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8596538?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8596538?buildTab=artifacts#/) on release-22.2 @ [94796359a4a73690b8e39484d8ecc515b64db77b](https://github.com/cockroachdb/cockroach/commits/94796359a4a73690b8e39484d8ecc515b64db77b):


```
=== RUN   TestMuxRangeFeedConnectsToNodeOnce
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/40ec041d13327ceafaefe721fe51f834/logTestMuxRangeFeedConnectsToNodeOnce492055702
    test_log_scope.go:79: use -show-logs to present logs inline
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1fca2e3]

goroutine 566163 [running]:
github.com/cockroachdb/cockroach/pkg/roachpb.(*RangeFeedEvent).GetValue(...)
	github.com/cockroachdb/cockroach/pkg/roachpb/bazel-out/k8-fastbuild/bin/pkg/roachpb/roachpb_go_proto_/github.com/cockroachdb/cockroach/pkg/roachpb/api.pb.go:23214
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).singleRangeFeed(0xc008470a00, {0x521c578?, 0xc0120c4e00?}, {{0xc0110c2500, 0x2, 0x8}, {0xc00f9d9ae8, 0x3, 0x8}}, {0x174132ad19cb8ec4, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:664 +0x1023
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).partialRangeFeed(0xc008470a00, {0x521c578, 0xc0120c4e00}, 0xc0120bfd10, 0xc0120acde0?, {{0xc0110c2500, 0x2, 0x8}, {0xc00f9d9ae8, 0x3, ...}}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:423 +0x8fb
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).RangeFeedSpans.func1.1({0x521c578?, 0xc0120c4e00?})
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:207 +0x117
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x25
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:74 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:71 +0xa5
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #96558 kv/kvclient/kvcoord: TestMuxRangeFeedConnectsToNodeOnce failed [C-test-failure O-robot branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMuxRangeFeedConnectsToNodeOnce.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvclient/kvcoord.TestMuxRangeFeedConnectsToNodeOnce [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8596541?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8596541?buildTab=artifacts#/) on release-22.2 @ [94796359a4a73690b8e39484d8ecc515b64db77b](https://github.com/cockroachdb/cockroach/commits/94796359a4a73690b8e39484d8ecc515b64db77b):


```
=== RUN   TestMuxRangeFeedConnectsToNodeOnce
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/40ec041d13327ceafaefe721fe51f834/logTestMuxRangeFeedConnectsToNodeOnce3383588020
    test_log_scope.go:79: use -show-logs to present logs inline
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x322261d]

goroutine 7897 [running]:
github.com/cockroachdb/cockroach/pkg/roachpb.(*RangeFeedEvent).GetValue(...)
	github.com/cockroachdb/cockroach/pkg/roachpb/bazel-out/k8-fastbuild/bin/pkg/roachpb/roachpb_go_proto_/github.com/cockroachdb/cockroach/pkg/roachpb/api.pb.go:23214
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).singleRangeFeed(0xc001da3400, {0x8c07270, 0xc002550000}, {{0xc004c58ce8, 0x3, 0x8}, {0xc005c98938, 0x4, 0x8}}, {0x174133504b92e2d8, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:664 +0x153d
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).partialRangeFeed(0xc001da3400, {0x8c07270, 0xc002550000}, 0xc0063e6b90, 0x17ebc85?, {{0xc004c58ce8, 0x3, 0x8}, {0xc005c98938, 0x4, ...}}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:423 +0xb3d
github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).RangeFeedSpans.func1.1({0x8c07270, 0xc002550000})
	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:207 +0x21f
github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1()
	github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:168 +0x4d
golang.org/x/sync/errgroup.(*Group).Go.func1()
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:74 +0x87
created by golang.org/x/sync/errgroup.(*Group).Go
	golang.org/x/sync/errgroup/external/org_golang_x_sync/errgroup/errgroup.go:71 +0x12f
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #96558 kv/kvclient/kvcoord: TestMuxRangeFeedConnectsToNodeOnce failed [C-test-failure O-robot branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMuxRangeFeedConnectsToNodeOnce.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go
+++ b/pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go
@@ -345,7 +345,7 @@ func (m *rangefeedMuxer) receiveEventsFromNode(ctx context.Context, ms *muxClien
 			// Normally, when ctx is done, we would receive streamErr above.
 			// But it's possible that the context was canceled right after the last Recv(),
 			// and in that case we must exit.
-			return nil
+			return ctx.Err()
 		case <-m.demuxLoopDone:
 			// demuxLoop exited, and so should we (happens when main context group completes)
 			return nil

